src/debugger/VariableFactory.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { VariableContainer, Variable } from "./types.js";
+import { ReferenceMap } from "./ReferenceMap.js";
+import { LexicalEnvironment, DeclarativeEnvironmentRecord } from "./../environment.js";
+import { Value, ConcreteValue, PrimitiveValue } from "./../values/index.js";
+
+export class VariableFactory {
+  constructor() {
+    this._referenceMap = new ReferenceMap();
+  }
+  _referenceMap: ReferenceMap<VariableContainer>;
+
+  createReference(value: VariableContainer): number {
```

Comment:
How do you handle if "scopes" command is called multiple times for the stop session?